bumble/rfcomm.py

```diff
@@ -549,8 +554,7 @@ def on_uih_frame(self, frame: RFCOMM_Frame) -> None:
             f'rx_credits={self.rx_credits}: {data.hex()}'
         )
         if data:
-            if self.sink:
-                self.sink(data)  # pylint: disable=not-callable
+            asyncio.create_task(self._dispatch_data(data))
```

Comment:
Creating a task for each packet may be a bit heavy. An alternative is to use an async queue, and a single task that waits for `opened`, then takes packets from the queue and forwards them to the sink.

Reply:
Waiting for opened as a task, but keeping other packets synchronously delivered might not (?) promise the order of packets. How about making a queue as a default sink, and let sink become a property - when it's set, dump all queued packets sink to the handler?

Reply:
I mean delivering packets synchornously to a queue, and has the async task pull from that queue. That guarantees the order.
Something like this: https://github.com/google/bumble/blob/22eb405fde31628c64ff1a1a7adc584bf140fe42/apps/speaker/speaker.py#L150
(in your case, the `pump_packets` task would first wait for `opened` before the `While`).

Reply:
The other model you implemented, with a queue to hold the packets until a sink is set, works fine too. It does require the user to be careful to not set the sink too early, but since this is for internal use, we an easily guarantee that.